src/validator/TraversalValidator.cpp

```diff
@@ -199,13 +199,14 @@ PlanNode* TraversalValidator::buildRuntimeInput(Starts& starts, PlanNode*& proje
 Expression* TraversalValidator::buildNStepLoopCondition(uint32_t steps) const {
     VLOG(1) << "steps: " << steps;
     // ++loopSteps{0} <= steps
-    auto loopSteps = vctx_->anonVarGen()->getVar();
-    qctx_->ectx()->setValue(loopSteps, 0);
+    if (LIKELY(loopSteps_.empty())) {
+        loopSteps_ = vctx_->anonVarGen()->getVar();
+    }
```

Comment:
Why not initialize this variable in constructor?

Reply:
done